targeteditor/src/main/scala/explore/targeteditor/RVInput.scala

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) 2016-2020 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package explore.targeteditor
+
+import cats.data.ValidatedNec
+import cats.effect.Async
+import cats.effect.ContextShift
+import cats.effect.IO
+import cats.implicits._
+import crystal.ViewF
+import crystal.react.implicits._
+import eu.timepit.refined.auto._
+import explore.components.ui.ExploreStyles
+import explore.model.formats._
+import explore.optics._
+import japgolly.scalajs.react._
+import japgolly.scalajs.react.vdom.html_<^._
+import lucuma.core.math.ApparentRadialVelocity
+import lucuma.core.math.RadialVelocity
+import lucuma.core.math.Redshift
+import lucuma.core.util.Display
+import lucuma.core.util.Enumerated
+import lucuma.ui.forms.EnumViewSelect
+import lucuma.ui.forms.FormInputEV
+import lucuma.ui.optics.ValidFormatInput
+import lucuma.ui.reusability._
+import monocle.macros.Lenses
+import react.common._
+import react.common.implicits._
+import react.semanticui.elements.label.LabelPointing
+
+final case class RVInput(
+  value:           ViewF[IO, Option[RadialVelocity]],
+  modify:          Option[RadialVelocity] => IO[Unit]
+)(implicit val cs: ContextShift[IO])
```

Comment:
There's a mechanism to access the `IO` typeclass instances from the context, so that we don't have to pass them around:

You can do inside the `render` method: `AppCtx.withCtx{ implicit ctx => ...}`. If you have `import explore.implicits._` then you will have implicit values for: `ContextShift[IO]`, `Timer[IO]`, `Logger[IO]`, and the GraphQL clients.

Reply:
Thanks I'll change that